bundle/edu.gemini.pot/src/main/scala/edu/gemini/spModel/target/env/BagsStatus.scala

```diff
@@ -0,0 +1,78 @@
+package edu.gemini.spModel.target.env
+
+import edu.gemini.shared.util.immutable.{Option => GOption}
+import edu.gemini.spModel.pio.{ParamSet, Pio, PioFactory}
+import edu.gemini.spModel.target.SPTarget
+import edu.gemini.spModel.rich.shared.immutable._
+
+// Possible BAGS statuses used by GuideProbeTargets to indicate the state of BAGS targets and lookups.
+sealed trait BagsStatus extends Cloneable {
+  val bagsStar: Option[SPTarget] = None
+  def bagsStarAsJava: GOption[SPTarget] = bagsStar.asGeminiOpt
+  override def clone: BagsStatus = this
+
+  def getParamSet(factory: PioFactory): ParamSet = {
+    val paramSet = factory.createParamSet(BagsStatus.BagsStatusParamSetName)
+    Pio.addParam(factory, paramSet, BagsStatus.BagsStatusParamName, id)
+    bagsStar.foreach { t =>
+      val bagsParamSet = factory.createParamSet(BagsStatus.BagsTargetParamSetName)
+      bagsParamSet.addParamSet(t.getParamSet(factory))
+    }
+    paramSet
+  }
+
+  def id: String
+}
+
+// TODO: For objects, we want to just return this.
+// TODO: For case class, we want to return an instance.
+
+// Note that we use BagsLookupPending as a default state when BAGS hasn't run or should be rerun, and the BagsManager
+// determines based on the observation if BAGS actually should be run.
+case object BagsLookupPending extends BagsStatus {
+  override val id = "BagsLookupPending"
+}
+case object BagsLookupRunning extends BagsStatus {
+  override val id = "BagsLookupRunning"
+}
```

Comment:
I hate to bring this up, but in light of Andy's findings about BAGS and program synchronization ...  Perhaps it isn't a good idea to store transient states in the model, if that's what we're doing (?).  Certainly the end result of the BAGS search is necessary in the model but updating to store intermediate results will mark the target node as modified even if in the end the search comes up with the same answer.  That in turn only increases the chance of bogus conflict reports.  Would it be possible to manage the state outside of the model and only store the final result, and only if it differs from what is already stored?  You may be doing it that way already?